slasher/server.go

```diff
@@ -0,0 +1,65 @@
+package slasher
+
+import (
+	"context"
+	"github.com/pkg/errors"
+
+	"github.com/gogo/protobuf/proto"
+	types "github.com/gogo/protobuf/types"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+	ethpb "github.com/prysmaticlabs/prysm/proto/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/slasher/db"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// Server defines a server implementation of the gRPC Slasher service,
+// providing RPC endpoints for retrieving slashing proofs for malicious validators.
+type Server struct {
+	slasherDb *db.Store
+	ctx       context.Context
+}
+
+// IsSlashableAttestation returns an attester slashing if the attestation submitted
+// is a slashable vote.
+func (ss *Server) IsSlashableAttestation(ctx context.Context, req *ethpb.Attestation) (*ethpb.AttesterSlashing, error) {
+	return nil, status.Error(codes.Unimplemented, "not implemented")
+}
+
+// IsSlashableBlock returns a proposer slashing if the block header submitted is
+// a slashable proposal.
+func (ss *Server) IsSlashableBlock(ctx context.Context, psr *ethpb.ProposerSlashingRequest) (*ethpb.ProposerSlashingResponse, error) {
+	//TODO(#3133): add signature validation
+	epoch := helpers.SlotToEpoch(psr.BlockHeader.Slot)
+	bha, err := ss.slasherDb.BlockHeader(epoch, psr.ValidatorIndex)
```

Comment:
maybe a better var name then `bha`

Reply:
thanks, changed

Reply:
the name still isnt changed , its still `bha`